### ISSUE TITLE
Does photoquery.py have duplicate code (lines 578 vs 583)?

### ISSUE THREAD

**Issue by mravery** [RhetTbull/osxphotos#1731]
**Describe the bug**
Starting in line 578 and ending 587, it looks like these two if blocks are exactly the same (i.e. you can delete one of them without affecting the logic).

**To Reproduce**
View the code at the above lines.

**Screenshots**
<img width="909" alt="Screen Shot 2024-11-04 at 8 21 12 PM" src="https://github.com/user-attachments/assets/75047e2a-8840-4f48-856a-2ccc34caefcb">

**Additional context**
I was just browsing the code and came upon this. I am not well-versed in python, nor am I well-versed in the code-base. There could be a legitimate reason for this.

**Comment by RhetTbull** [comment]
You're right! Thanks for catching this. Not a bug technically but the code is unnecessary and doesn't actually do anything.

**Comment by RhetTbull** [comment]
Looking at this code it should really be refactored to be simplified. If I change all the options to match the name of the PhotoInfo property, e.g. `option.isreference` and `option.not_isreference` then much of the code could be greatly simplified. (Today, some of the options have a different name than the PhotoInfo property, e.g. `option.is_reference`). We could then do something like this (after accounting for the text search properties like `description`:


```python
def filter_photos(options: Options, photos: list[PhotoInfo]) -> list[PhotoInfo]:
    filtered_photos = photos
    for field in options.fields():
        if field.startswith("not_"):
            prop = field[4:]
            filtered_photos = [photo for photo in filtered_photos if not getattr(photo, prop, None)]
        else:
            filtered_photos = [photo for photo in filtered_photos if getattr(photo, field, None)]
    
    return filtered_photos
```

This would also make adding new options easier. There are a few special cases that would need to be handled.
